app/javascript/mastodon/features/direct_timeline/components/conversations_list.jsx

```diff
@@ -1,77 +1,70 @@
 import PropTypes from 'prop-types';
+import { useRef, useCallback } from 'react';
 
-import ImmutablePropTypes from 'react-immutable-proptypes';
-import ImmutablePureComponent from 'react-immutable-pure-component';
+import { useSelector, useDispatch } from 'react-redux';
 
 import { debounce } from 'lodash';
 
-import ScrollableList from '../../../components/scrollable_list';
-import ConversationContainer from '../containers/conversation_container';
+import { expandConversations } from 'mastodon/actions/conversations';
+import ScrollableList from 'mastodon/components/scrollable_list';
 
-export default class ConversationsList extends ImmutablePureComponent {
+import { Conversation } from './conversation';
 
-  static propTypes = {
-    conversations: ImmutablePropTypes.list.isRequired,
-    scrollKey: PropTypes.string.isRequired,
-    hasMore: PropTypes.bool,
-    isLoading: PropTypes.bool,
-    onLoadMore: PropTypes.func,
-  };
-
-  getCurrentIndex = id => this.props.conversations.findIndex(x => x.get('id') === id);
+export const ConversationsList = ({ scrollKey, ...other }) => {
+  const listRef = useRef();
+  const conversations = useSelector(state => state.getIn(['conversations', 'items']));
+  const isLoading = useSelector(state => state.getIn(['conversations', 'isLoading'], true));
+  const hasMore = useSelector(state => state.getIn(['conversations', 'hasMore'], false));
+  const dispatch = useDispatch();
 
-  handleMoveUp = id => {
-    const elementIndex = this.getCurrentIndex(id) - 1;
-    this._selectChild(elementIndex, true);
-  };
+  const handleMoveUp = useCallback(id => {
+    const elementIndex = conversations.findIndex(x => x.get('id') === id) - 1;
+    focusChild(elementIndex, true);
+  }, [conversations]);
 
-  handleMoveDown = id => {
-    const elementIndex = this.getCurrentIndex(id) + 1;
-    this._selectChild(elementIndex, false);
-  };
+  const handleMoveDown = useCallback(id => {
+    const elementIndex = conversations.findIndex(x => x.get('id') === id) + 1;
+    focusChild(elementIndex, false);
+  }, [conversations]);
 
-  _selectChild (index, align_top) {
-    const container = this.node.node;
-    const element = container.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
+  const focusChild = (index, alignTop) => {
+    const node = listRef.current;
+    const element = node.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
```

Comment:
This is incorrect and fails with `node.querySelector` is not a function.

I believe this should be changed to:
```suggestion
    const node = listRef.current.node;
    const element = node.querySelector(`article:nth-of-type(${index + 1}) .focusable`);
```